### PR TITLE
add namespace attribute in keycloakx objects

### PR DESCRIPTION
**Pull request by bovy89**
Signed-off-by: bovy89 <bovy89@gmail.com>

### CODE REVIEW

**Comment by PSanetra** [comment]
Hi @bovy89, can you tell us more about the use case for this change? I think at least the changes related to setting the namespace fields to `.Release.Namespace` seem to be unnecessary.

**Comment by bovy89** [comment]
Some tools are using helm template (`helm template -n <namespace>`) to generate a static manifest. That change is also helpful to include namespace info in order to exclude an entire namespace from automated testing (we are using trivy rego rules) without using anything else (i.e. kustomize).

Some honorable mentions of charts that are doing that:
- https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack/templates/prometheus
- https://github.com/kubernetes/ingress-nginx/tree/main/charts/ingress-nginx

File: charts/keycloakx/templates/prometheusrule.yaml [hunk 1]
```diff
@@ -4,6 +4,9 @@ apiVersion: monitoring.coreos.com/v1
 kind: PrometheusRule
 metadata:
   name: {{ include "keycloak.fullname" $ }}
+  {{- with .namespace }}
+  namespace: {{ . }}
```
Comment (mircohaug) [inline charts/keycloakx/templates/prometheusrule.yaml:8]: This needs to default explicitly to `{{ .Release.Namespace }}` as well

**Review by mircohaug — CHANGES_REQUESTED** [review]
Thanks for the contribution, please also add `{{ .Release.Namespace }}` as default when for the resources where namespace overwrites are possible. Namely the `PrometheusRule` you added the overwrite for but also the `ServiceMonitor` resource where this overwrite already exists

**Comment by bovy89** [comment]
Any update?

**Comment by grieshaber** [comment]
hey @bovy89 thanks for this PR. We had some changes in the maintainer team recently and lost a little bit of track.
I'll check your chages real quick and merge the PR if nothing comes up.

**Review by grieshaber — APPROVED** [review]

**Pull request merged**